File: bike-rental-golang-api/pkg/list/bike.go
package list

import (
	"github.com/google/uuid"
	"github.com/nicoflink/bike-rental/pkg/geo"
)

// Bike is bike used for the view in the bike-app
// Rented: Bike is rented
// RentedByUser: Bike is rented by the current user requesting the resource.
type Bike struct {
	ID           uuid.UUID
	Name         string
	Location     geo.Coordinates
	Rented       bool
	RentedByUser bool
}

// BikeLocationUpdate used for location update of the bike.
type BikeLocationUpdate struct {
	BikeID   uuid.UUID
	Location geo.Coordinates
}

File: bike-map-app/src/app/bike.ts
// Reprensentation of a bike in the fronend
export interface Bike {
    id: string,
    name: string,
    location: {
        latitude: number,
        longitude: number 
    },
    rented: boolean,
    returnable: boolean // returnable is true if renter of the bike is the current user
}

// Information needed to update the location of the bike.
export interface BikeLocationUpdate {
    bikeID: string,
    location: {
        latitude: number,
        longitude: number 
    },
}